docs/concepts/style_guide.md

```diff
@@ -0,0 +1,206 @@
+reader that the system state will be restored.
+
+**Good:**
+```ruby
+test_name 'The source attribute' do
+
+  target_file_on_nix = '/tmp/source_attr_test'
+  teardown do
+    on(hosts[0], "rm #{target_file_on_nix}", :accept_all_exit_codes => true)
+  end
+  ...
+end
+```
+
+Teardowns are at the mercy of the scoping of the variables necessary to perform
+the restoration of the system. This fact means that additional teardown steps
+will need to be added within the scope necessary to do their job. Effort should
+be taken to make the teardown steps prominent and readable so that it can be
+confirmed, via the logs, that the system has been restored.
+
+Teardown steps registered outside of tests should use
+[`#step`](http://www.rubydoc.info/github/puppetlabs/beaker/Beaker/DSL/Structure#step-instance_method)
+to document and log what they are doing.
+
+## Acceptable Exit Codes
+
+When using the Beaker
+[`on`](http://www.rubydoc.info/github/puppetlabs/beaker/Beaker/DSL/Helpers/HostHelpers#on-instance_method)
+method, the default setting is that only an exit code of 0 (zero) will not
+trigger an error. When other exit codes are acceptable, the
+`:acceptable_exit_codes` key with an array of exit codes should be passed in the
+options hash to `#on`. If 0 (zero) is the only acceptable exit code, then the
+`:acceptable_exit_codes` symbol must not be used.
+
+**Good:**
+
+- Single 0 exit code allowed
+```ruby
+on(host, "rm #{file_to_rm}")
+```
+- Single non-0 exit code allowed
+```ruby
+on(host, "rm #{file_to_rm}", :acceptable_exit_codes => 1)
+```
+- Multiple exit codes allowed
+```ruby
+on(host, "rm #{file_to_rm}", :acceptable_exit_codes => [0,1])
+```
+- Any exit code allowed
+```ruby
+on(host, "rm #{file_to_rm}", :accept_all_exit_codes => true)
+```
+
+In the last case, when any exit code is allowed, one must follow-up with a valid
+assertion test.
+
+If an exit_code outside of 0 is expected, one must use acceptable_exit_codes so
+the test will fail on the proper assertion and not error at that command.  Allow
+only the minimum expected set of exit codes unless coverage is provided by
+subsequent assertions.
+
+## Outcomes
+
+When to use each, how to format the message:
+
+### `fail_test`, `pass_test`, & `skip_test`
+
+These can be used anywhere in a test to exit early. A
+[`skip_test`](http://www.rubydoc.info/github/puppetlabs/beaker/Beaker/DSL/Outcomes#fail_test-instance_method)
+between two assertions, for instance, will run the first assertion, raise an
+exception for `skip_test`, run teardown, and then exit as we expect. The same is
+true for
+[fail_test](http://www.rubydoc.info/github/puppetlabs/beaker/Beaker/DSL/Outcomes#fail_test-instance_method).
+
+[pass_test](http://www.rubydoc.info/github/puppetlabs/beaker/Beaker/DSL/Outcomes#fail_test-instance_method)
+should not be used unless fast-passing in a conditional branch.
+
+### Skipping Tests
+
+Skipping tests can be used, for instance, when they are temporarily failing or
```

Comment:
@cowofevil I've changed the ordering to put the `expect_failure` section first. I've actually just put it up above the "`fail_test`, `pass_test`, ..." section to show this preference.

I didn't add it to that same section though, because it didn't seem to fit the text in the section as well as the others. Specifically, the methods mentioned are used solely to exit early with a certain status. They don't take blocks, or do anything complex, they just quit, whereas the `expect_failure` method is more complex and depends on the body you pass it. Does this make sense?